### ISSUE TITLE
Set up 2 mac code signing machines

### ISSUE THREAD

**Issue by zanderso** [flutter/flutter#42249]
Set up 2 mac code signing machines.

# TODO
- [ ] Add the code-signing machine to luci.flutter.prod
- [ ] Automate the signing process
- [ ] Reprovision two new machines and add them to internal infrastructure

**Comment by digiter** [comment]
This is about shipping the binaries, not development in the device lab.

**Comment by CaseyHillers** [comment]
We'll start by adding the machine to a separate pool. Then we'll need to sort out the recipes relying on this pool.

Another issue is having the framework waiting on the engine builds being signed before triggering.

**Comment by godofredoc** [comment]
I'll try to fix the salt states to enable the signer machine in one of our pools.

**Comment by CaseyHillers** [comment]
The current state:

- flutter-devicelab-mac-signing-1 was added to the DeviceLab, but salt failed to install on it when I tried in ~January
- flutter-devicelab-mac-signing-2 is at my desk, and needs to be reimaged before it can be added to the lab

**Comment by yusuf-goog** [comment]
Taking a look at flutter-devicelab-mac-signing-1

**Comment by yusuf-goog** [comment]
I've hooked up flutter-devicelab-mac-signing-1 to salt dev master and applied states. Its failing the some states (machine tokend, certs, bot related), but should be in a basic workable state. Will need to tweak the salt states to setup for a not-a-swarming-bot-mac-host.

**Comment by yusuf-goog** [comment]
Adding @XilaiZhang  who will be working on configuring the code signing hosts.

**Comment by yusuf-goog** [comment]
@zanderso the TODO list in the description are not applicable to this bug??

**Comment by christopherfujino** [comment]
> @zanderso the TODO list in the description are not applicable to this bug??

@CaseyHillers added those TODOs

**Comment by godofredoc** [comment]
Updated the list

**Comment by yusuf-goog** [comment]
Applied flutter.code_signing state to flutter-devicelab-mac-signing-1

**Comment by godofredoc** [comment]
code_signing in this machines will most likely be manually configured.

**Comment by keyonghan** [comment]
From triage: @XilaiZhang Could you help update with latest status?

**Comment by XilaiZhang** [comment]
yes, currently we are trying to get manual signing to work on the bot. Once we figure out manual signing, we can start getting the process automated.

**Comment by godofredoc** [comment]
@XilaiZhang are you still blocked on the salt confis?

**Comment by XilaiZhang** [comment]
Yeah kind of, currently working on making salt state apply to work, I am going to the lab to try one more time this afternoon. Keyong will be back tomorrow and we can do more tests. Sorry for the delay, not sure if workday propagated the information, but yeah I went to spine center this morning

**Comment by XilaiZhang** [comment]
Tested just now and it’s the same error, master rejected minion’s public key. Should probably get solved when keyong is back tomorrow. I will head back to 1808 now.

**Comment by godofredoc** [comment]
The bot we are trying to setup is https://chromium-swarm.appspot.com/bot?id=flutter-devicelab-mac-signing-3

**Comment by godofredoc** [comment]
@XilaiZhang can this be closed?

**Comment by XilaiZhang** [comment]
Yep signing-3 is fully setup. I just logged into signing-2, it looks like it is missing Xcode, certs and signing salt state. I downloaded and applied all the pre-reqs just now. will close this issue for now.
